lib/screens/home/carts/modals/select_user_screen.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:shop_app/blocs/index.dart';
+import 'package:shop_app/localization/index.dart';
+import 'package:shop_app/models/index.dart';
+import 'package:shop_app/router/index.dart';
+
+@RoutePage()
+class SelectUserScreen extends StatelessWidget implements AutoRouteWrapper {
+  const SelectUserScreen({super.key});
+
+  @override
+  Widget wrappedRoute(BuildContext context) {
+    context.read<UsersBloc>().load();
+
+    return this;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    EasyLocalization.of(context);
+
+    final usersBloc = context.read<UsersBloc>();
+
+    return Material(
+      child: RefreshIndicator(
+        onRefresh: () {
+          return usersBloc.loadAsyncFuture();
+        },
+        child: CustomScrollView(
+          slivers: [
+            SliverAppBar(
+              pinned: true,
+              floating: true,
+              automaticallyImplyLeading: false,
+              title: TextField(
+                decoration: const InputDecoration(
+                  filled: true,
+                  fillColor: Colors.white70,
+                  suffixIcon: Icon(Icons.search),
+                ),
+                onSubmitted: (value) {
+                  usersBloc.search(value);
```

Comment:
You are using the same bloc which operates in UsersScreen, so when you call search -> UsersScreen will be filtered too, therefore you have to incapsulate this logic.
If you want to reuse UsersBloc create Provider for this screen, which will have data, visibleData and implement search along with inside this provider you can call usersBloc.load